apis/v1alpha1/cluster_bpf_application_types.go

```diff
@@ -71,7 +71,11 @@ type ClBpfApplicationProgram struct {
 	// Name is the name of the function that is the entry point for the BPF
 	// program
 	Name string `json:"name"`
-
+	// AutoDiscovery when enabled, the agent process monitors the creation and deletion of interfaces,
+	// automatically attaching eBPF hooks to newly discovered interfaces in both directions.
+	//+kubebuilder:default:=false
+	// +optional
+	AutoDiscovery bool `json:"autoDiscovery"`
```

Comment:
This should probably be called something like "InterfaceAutoDiscovery" to make it clear that it applies to interfaces.  Also, will you need some additional configuration interfaces to ignore, or interface name patterns?

Reply:
yes we need interface list to ignore at min we can get fancy with regex matching later but I think for GA we need to enable it and make sure the interface is not in the ignore list